controllers/dexserver_controller.go

```diff
@@ -577,6 +586,53 @@ func (r *DexServerReconciler) syncDeployment(dexServer *authv1alpha1.DexServer,
 	return nil
 }
 
+// Copy a secret from its original namespace into the Dex Server namespace
+func (r *DexServerReconciler) copySecretToDexServerNamespace (dexServer *authv1alpha1.DexServer, secretRef corev1.SecretReference, ctx context.Context) (error) {
+	log := ctrllog.FromContext(ctx)
+
+	// Secret to copy from
+	originalSecret := &corev1.Secret{}
+	if err := r.Client.Get(context.TODO(),
+		client.ObjectKey{Name: secretRef.Name, Namespace: secretRef.Namespace},
+		originalSecret); err != nil {
+		log.Error(err, "Error retrieving secret", "name", secretRef.Name)
+		return err
+	}
+	// Add label to this secret so that the secret can be watched for updates
+	checkAndAddLabelToSecret(originalSecret, r, ctx)	
+
+	// Secret to copy into (in the dex server namespace)
+	secretInDexServerNS := &corev1.Secret{}
+
+	if err := r.Client.Get(context.TODO(), client.ObjectKey{Name: originalSecret.Name, Namespace: dexServer.Namespace},
```

Comment:
Wonder if the name of the new secret should not be a concatenation of the original secret ns and name to avoid colision.